plugins/janus_recordplay.c

```diff
@@ -448,6 +450,143 @@ typedef struct janus_recordplay_session {
 static GHashTable *sessions;
 static janus_mutex sessions_mutex = JANUS_MUTEX_INITIALIZER;
 
+/*! \brief Helper function to filter and map \ref janus_sdp_attribute list to
+ * (int*)id:(char*)info extmap hash table
+ * \param extmaps Hash table to contain filtered and mapped extmaps
+ * \param attributes List of sdp media attributes
+ */
+static void janus_recordplay_map_extmaps(GHashTable *extmaps, const GList *attributes) {
+	if(extmaps == NULL)
+		return;
+	g_hash_table_remove_all(extmaps);
+	for(; attributes != NULL; attributes = attributes->next) {
+		janus_sdp_attribute *attr = (janus_sdp_attribute *)attributes->data;
+		if(attr->name == NULL || attr->value == NULL) {
+			JANUS_LOG(LOG_WARN, "Extmap has null fields\n");
+			continue;
+		}
+		if(!strcasecmp(attr->name, "extmap")) {
+			size_t id_len = strcspn(attr->value, " /");
+			if(!id_len || id_len == strlen(attr->value)) {
+				JANUS_LOG(LOG_WARN, "Extmap 'extmap:%s' is malformed\n", attr->value);
+				continue;
+			}
+			uint32_t *ext_id = g_new(uint32_t, 1);
+			if(!janus_string_to_uint32(attr->value, ext_id)) {
+				g_hash_table_insert(extmaps, ext_id, g_strdup(attr->value+id_len));
+			} else {
+				JANUS_LOG(LOG_WARN, "Extmap has invalid id\n");
+				g_free(ext_id);
+			}
+		}
+	}
+}
+
+/*! \brief Helper function to copy extmaps from one table to other
+ * \param dst Destination table
+ * \param src Source table
+ */
+static void janus_recordplay_copy_extmaps(GHashTable *dst, GHashTable *src) {
+	if(dst == NULL)
+		return;
+	g_hash_table_remove_all(dst);
+	if(src == NULL)
+		return;
+	GHashTableIter iter;
+	uint32_t *ext_id;
+	char *ext_info;
+	g_hash_table_iter_init(&iter, src);
+	while(g_hash_table_iter_next(&iter, (gpointer *)&ext_id, (gpointer *)&ext_info)) {
+		if(ext_id == NULL || ext_info == NULL) {
+			JANUS_LOG(LOG_WARN, "Extmap has null fields\n");
+			continue;
+		}
+		g_hash_table_insert(dst, janus_uint32_dup(*ext_id), g_strdup(ext_info));
+	}
+}
+
+/*! \brief Helper function to update extmap attributes in media
+ * \param media Parsed SDP mline
+ * \param extmaps Table of extmaps to add to media
+ */
+static void janus_recordplay_regenerate_extmaps(janus_sdp_mline *media, GHashTable *extmaps) {
+	if(media == NULL || extmaps == NULL)
+		return;
+	GHashTable *id2info = g_hash_table_new(g_int_hash, g_int_equal);
+	GHashTable *info2id = g_hash_table_new(g_str_hash, g_str_equal);
+	GHashTable *generated = g_hash_table_new_full(g_int_hash, g_int_equal, g_free, g_free);
+	GHashTableIter iter;
+	uint32_t *ext_id, i;
+	char *ext_info;
+	GQueue *indices = g_queue_new();
+	/* Add saved extmaps */
+	g_hash_table_iter_init(&iter, extmaps);
+	while(g_hash_table_iter_next(&iter, (gpointer *)&ext_id, (gpointer *)&ext_info)) {
+		if(ext_id == NULL || ext_info == NULL) {
+			JANUS_LOG(LOG_WARN, "Extmap has null fields\n");
+			continue;
+		}
+		JANUS_LOG(LOG_INFO, " > %d:%s\n", *ext_id, ext_info);
```

Comment:
I'd make this log more specific than just `id > info`.
Also consider lowering its level to verbose.

Reply:
This is one of debug logs I used for testing, it shouldn't be here. Thanks for spotting this.